### DISCUSSION TITLE
ALS Refactored with Smooth Sync

### DISCUSSION THREAD

**Discussion by emrehan98** [Sixze/ALS-Refactored#20] [Ideas]
Hey, Sixze. 

I've been trying to implement Smooth Sync with ALS Refactored. Smooth Sync is a plugin that simply smooths player location over network so even in extreme movements, such as climbing, flying etc. it gives you very smooth results over other clients and server with no effort so you don't really worry about custom movement and jittering problems. 

But in order to make it work, you need to disable **"Replicate Movement".** 

When I disable "Replicate Movement",  ALS Refactored still functions pretty well but one thing that doesn't work is turn-in-place and I couldn't find the reason. I implemented my own variable to fix that problem but it wasn't the perfect solution as the result was jittery. (Animations are jittering, not the rotation.) I'm not CPP programmer, I can just read the code and make sense of it, do some little stuff so my solution is obviously not the best one.

https://user-images.githubusercontent.com/19621279/134226373-9c2f0fff-0224-4924-b97b-6463f154f356.mp4

Here's what I did.

https://user-images.githubusercontent.com/19621279/134226927-371fb44b-a388-4564-aeb7-3ea9771e5512.mp4

I guess when I disable "Replicate Movement", ViewRotation no longer replicated in other clients so only owner can see animations work properly. I'm not sure if it's out of scope but it would be a huge help to make ViewRotation "Replicate Movement" independent so I could use both plugins. But if you can point me a direction that would be amazing too.

**Comment by emrehan98** [comment]
I made it work, so the problem was not using `IsLocallyControlled`. I thought `(GetLocalRole() >= ROLE_AutonomousProxy)` doing the same job but I was wrong.

So I discarded everything again, returned back to original ALS Refactored.

1 - I added "Server, Unreliable" to "SetViewRotation" in AlsCharacter.h
2 - Added "_Implementation" to end of the "SetViewRotation" in AlsCharacter.cpp.
3 - I changed the RefreshView function.
![resim](https://user-images.githubusercontent.com/19621279/134239029-4f2d5c69-6aed-4091-b865-680ee98e97d5.png)

Now head rotations and turn in place works properly.

**Comment by Sixze** [comment]
I do not replicate the view rotation directly from client to server because the movement component already does it itself, but it looks like when the movement is not replicated, the view rotation is not replicated either.

The best option would be to rely on the replication of the movement component when `bReplicateMovement == true`, and, on the other hand, manually invoke the RPC as you do when `bReplicateMovement == false`.

I'll fix it soon in future commit.

**Comment by Sixze** [comment]
Fixed in 1920132678bb6dbb788e39ec571ce20977a7231d.

**Reply by emrehan98** [reply]
Thank you so much!
